Modules/Loadable/Markups/MRML/vtkMRMLMarkupsNode.cxx

```diff
@@ -1286,22 +1286,50 @@ void vtkMRMLMarkupsNode::GetRASBounds(double bounds[6])
   vtkMath::UninitializeBounds(bounds);
   int numberOfMarkups = this->GetNumberOfMarkups();
   if (numberOfMarkups == 0)
-  {
+    {
     return;
-  }
+    }
   double markup_RAS[4] = { 0, 0, 0, 1 };
   this->GetMarkupPointWorld(0, 0, markup_RAS);
+
   bounds[0] = bounds[1] = markup_RAS[0];
   bounds[2] = bounds[3] = markup_RAS[1];
   bounds[4] = bounds[5] = markup_RAS[2];
   for (int i = 1; i < numberOfMarkups; i++)
-  {
+    {
     this->GetMarkupPointWorld(i, 0, markup_RAS);
     if (markup_RAS[0] < bounds[0]) { bounds[0] = markup_RAS[0]; }
     if (markup_RAS[0] > bounds[1]) { bounds[1] = markup_RAS[0]; }
     if (markup_RAS[1] < bounds[2]) { bounds[2] = markup_RAS[1]; }
     if (markup_RAS[1] > bounds[3]) { bounds[3] = markup_RAS[1]; }
     if (markup_RAS[2] < bounds[4]) { bounds[4] = markup_RAS[2]; }
     if (markup_RAS[2] > bounds[5]) { bounds[5] = markup_RAS[2]; }
-  }
+    }
+}
+
+//---------------------------------------------------------------------------
+void vtkMRMLMarkupsNode::GetNodeBounds(double bounds[6])
+{
+  vtkMath::UninitializeBounds(bounds);
+  int numberOfMarkups = this->GetNumberOfMarkups();
+  if (numberOfMarkups == 0)
+    {
+    return;
+    }
+  double markup_RAS[4] = { 0, 0, 0, 1 };
+  this->GetMarkupPoint(0, 0, markup_RAS);
```

Comment:
Markup fiducials only have one point per markup, but in the future one markup can consist of multiple points, so it's not enough to just check the first one.
If you implement GetBounds in MarkupFiducial node then there you may check only the first point (it'll be faster than the generic vktMRMLMarkupsNode GetBounds, because it doesn' have to check how many points are there in each markup and iterate through them).
So,
1. move this GetBounds() implementation into vtkMRMLMarkupFiducialNode, and
2. Either don't implement GetBounds for vtkMRMLMarkupsNode (leave it to child classes) or implement it properly (iterate through all points in all markups)